src/rbindlist.c

```diff
@@ -310,10 +310,21 @@ SEXP rbindlist(SEXP l, SEXP usenamesArg, SEXP fillArg, SEXP idcolArg)
       if (firsti==-1) { firsti=i; firstw=w; firstCol=thisCol; }
       else {
         if (!factor && !int64) {
-          if (!R_compute_identical(PROTECT(getAttrib(thisCol, R_ClassSymbol)),
-                                   PROTECT(getAttrib(firstCol, R_ClassSymbol)),
-                                   0)) {
-            error(_("Class attribute on column %d of item %d does not match with column %d of item %d."), w+1, i+1, firstw+1, firsti+1);
+          SEXP thisClass = PROTECT(getAttrib(thisCol, R_ClassSymbol));
```

Comment:
@shrektan can you comment on if we need to do anything with encoding here? class can potentially be non-UTF8 right?

Reply:
In theroy, yes... and if not native encoded, the message would become garbage letters (the errors and warnings API of R expects native encoding)... However, I don't think anyone would actually use a non-ASCII string as the class name... 

Anyway, if this is needed, we should add the translation (making the native encoded string) in `concatCharVec()` . I mean, adding an extra argument `encoding` ( 0: do nothing 1: to utf-8 2: to native encoding ) there.

Reply:
OK, as long as it won't crash R, I think it's fine to leave that for future users to report :smiley: 

My guess is "ASCII in class name" is a broader assumption in the source code, can't think of any related issues I've seen.

Reply:
For what it's worth, I played around with this, but it didn't quite work the way I expected.
![image](https://user-images.githubusercontent.com/1295956/100535514-c998f200-326d-11eb-9dad-6e6e72208cef.png)
  

Using `CHAR()` or `Rf_translateCharUTF8()` gives this result:
![image](https://user-images.githubusercontent.com/1295956/100535466-51323100-326d-11eb-8c8a-42b3c0a1e130.png)

Using `Rf_translateChar()` or `Rf_translateChar0()` gives this:
![image](https://user-images.githubusercontent.com/1295956/100535479-79ba2b00-326d-11eb-983f-ee26faf193df.png)

If I try to write a test it fails either way:
![image](https://user-images.githubusercontent.com/1295956/100535870-cce1ad00-3270-11eb-9149-0e4f5fb9e7ef.png)
![image](https://user-images.githubusercontent.com/1295956/100535912-234eeb80-3271-11eb-89ac-1efa73892f3f.png)


Simple errors don't seem to display correctly either:
![image](https://user-images.githubusercontent.com/1295956/100535636-ca7e5380-326e-11eb-9b5a-f3ff7cf696dc.png)

🤷‍♂️